probe/kubernetes/client.go

```diff
@@ -24,6 +24,8 @@ import (
 	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
 )
 
+const errResourceNotFound = "the server could not find the requested resource"
```

Comment:
Can we do better? I suspect the error that is coming back in this case is a [`StatusError`](https://godoc.org/k8s.io/apimachinery/pkg/api/errors#StatusError), implementing [`APIStatus`](https://godoc.org/k8s.io/apimachinery/pkg/api/errors#APIStatus), which ought to return [`StatusReasonNotFound`](https://godoc.org/k8s.io/apimachinery/pkg/apis/meta/v1#StatusReason).